userspace/libpman/src/lifecycle.c

```diff
@@ -28,6 +29,43 @@ int pman_open_probe() {
 	return 0;
 }
 
+int pman_prepare_progs_before_loading() {
+	/*
+	 * Probe required features for each bpf program, as requested
+	 */
+	for(int ev = 0; ev < PPM_EVENT_MAX; ev++) {
+		int num_skipped = 0;
+		int idx = 0;
+		event_prog_t *progs = event_prog_table[ev];
+		for(; idx < MAX_FEATURE_CHECKS && progs[idx].name != NULL; idx++) {
```

Comment:
[nit] Some reordering here might help make it easier to read
```suggestion
		event_prog_t *progs = event_prog_table[ev];
		int num_skipped = 0;
		for(int idx = 0; idx < MAX_FEATURE_CHECKS && progs[idx].name != NULL; idx++) {
```